main/src/libraries/wrdac/src/subsystems/subSystem_ARE.cpp

```diff
@@ -325,6 +325,52 @@ bool wysiwyd::wrdac::SubSystem_ARE::push(const yarp::sig::Vector &targetUnsafe,
 
 }
 
+bool wysiwyd::wrdac::SubSystem_ARE::pointfar(const yarp::sig::Vector &targetUnsafe, const yarp::os::Bottle &options, const std::string &sName)
+{
+    if (ABMconnected)
+    {
+        std::list<std::pair<std::string, std::string> > lArgument;
+        lArgument.push_back(std::pair<std::string, std::string>(targetUnsafe.toString().c_str(), "vector"));
+        lArgument.push_back(std::pair<std::string, std::string>(options.toString().c_str(), "options"));
+        lArgument.push_back(std::pair<std::string, std::string>("pointfar", "predicate"));
+        lArgument.push_back(std::pair<std::string, std::string>(sName, "object"));
+        lArgument.push_back(std::pair<std::string, std::string>("iCub", "agent"));
+        lArgument.push_back(std::pair<std::string, std::string>(m_masterName, "provider"));
+        lArgument.push_back(std::pair<std::string, std::string>("ARE", "subsystem"));
+        SubABM->sendActivity("action", "pointfar", "action", lArgument, true);
+    }
+
+    yarp::os::Bottle bCmd;
+    bCmd.addVocab(yarp::os::Vocab::encode("pfar"));
+
+    yarp::sig::Vector target=targetUnsafe;
+    yarp::os::Bottle opt=options;
+    selectHandCorrectTarget(opt,target);
+    target=applySafetyMargins(target);
+    appendCartesianTarget(bCmd,target);
+    bCmd.append(opt);
+
```

Comment:
Better to do:
```cpp
yarp::sig::Vector target=applySafetyMargins(targetUnsafe);
appendCartesianTarget(bCmd,target);
bCmd.append(opt);
```
We don't have to apply IOL corrections to far away targets. Then, the hand is selected internally by `ARE`.